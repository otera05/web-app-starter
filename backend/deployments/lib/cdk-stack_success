import * as cdk from "@aws-cdk/core";
import * as ec2 from "@aws-cdk/aws-ec2";
import * as ecs from "@aws-cdk/aws-ecs";
import * as ecr from "@aws-cdk/aws-ecr";
import * as iam from "@aws-cdk/aws-iam";
import * as elbv2 from "@aws-cdk/aws-elasticloadbalancingv2";
import * as route53 from "@aws-cdk/aws-route53";
import * as route53Targets from "@aws-cdk/aws-route53-targets";
import { NetworkMode } from "@aws-cdk/aws-ecs";
import { Stage } from "@aws-cdk/core";

interface subnetConfig {
  subnetId: string;
  availabilityZone: string;
  routeTableId: string;
}

interface securityGroupConfig {
  id: string;
  name: string;
}

interface networkConfig {
  vpcName: string;
  publicSubnets: subnetConfig[];
  securityGroup: securityGroupConfig;
  targetGroupArn: string;
}
interface containerConfig {
  hostPort: number;
  containerPort: number;
  memoryLimitMiB: number;
  cpu: number;
  desiredCount: number;
  imageName: string;
  env: { [key: string]: string };
  assignPublicIp: boolean;
}

interface albConfig {
  id: string;
  name: string;
}

interface healthCheckConfig {
  statusCode: string;
}

interface targetGroupConfig {
  healthCheck: healthCheckConfig;
  port: number;
  name: string;
}

interface listenerHttpConfig {
  id: string;
  port: number;
}

interface listenerHttpsConfig {
  id: string;
  port: number;
  certificateArn: string;
}

interface listenerConfig {
  http: listenerHttpConfig;
  https: listenerHttpsConfig;
}

interface loadBalancerConfig {
  alb: albConfig;
  targetGroup: targetGroupConfig;
  listener: listenerConfig;
}

interface hostedZoneConfig {
  id: string;
  domainName: string;
}

interface route53Config {
  id: string;
  fqdn: string;
  hostedZone: hostedZoneConfig;
}

export interface StageContext {
  description: string;
  network: networkConfig;
  container: containerConfig;
  loadBalancer: loadBalancerConfig;
  route53: route53Config;
}

export class CdkStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    const env: string = this.node.tryGetContext("env") || "Stg";
    const revision: string = this.node.tryGetContext("revision") || "";
    const tag: string = this.node.tryGetContext("tag") || "";

    const context: StageContext =
      env == "Test"
        ? JSON.parse(this.node.tryGetContext(env))
        : this.node.tryGetContext(env);

    this.validateEnvironment(env, revision, context);

    // Create a cluster
    const vpc = new ec2.Vpc(this, "Vpc", { maxAzs: 2 });

    const cluster = new ecs.Cluster(this, "EcsCluster", { vpc });
    cluster.addCapacity("DefaultAutoScalingGroup", {
      instanceType: ec2.InstanceType.of(
        ec2.InstanceClass.T2,
        ec2.InstanceSize.MICRO
      ),
    });

    // Create Task Definition
    const taskDefinition = new ecs.Ec2TaskDefinition(this, "TaskDef");
    const container = taskDefinition.addContainer("web", {
      image: ecs.ContainerImage.fromRegistry("amazon/amazon-ecs-sample"),
      memoryLimitMiB: 256,
    });

    container.addPortMappings({
      containerPort: 80,
      hostPort: 8080,
      protocol: ecs.Protocol.TCP,
    });

    // Create Service
    const service = new ecs.Ec2Service(this, "Service", {
      cluster,
      taskDefinition,
    });

    // Create ALB
    const lb = new elbv2.ApplicationLoadBalancer(this, "LB", {
      vpc,
      internetFacing: true,
    });
    const listener = lb.addListener("PublicListener", { port: 80, open: true });

    // Attach ALB to ECS Service
    listener.addTargets("ECS", {
      port: 80,
      targets: [
        service.loadBalancerTarget({
          containerName: "web",
          containerPort: 80,
        }),
      ],
      // include health check (default is none)
      healthCheck: {
        interval: cdk.Duration.seconds(60),
        path: "/health",
        timeout: cdk.Duration.seconds(5),
      },
    });

    this.createARecordWithAlb(context.route53, lb);
  }

  private validateEnvironment(
    env: string,
    revision: string,
    context: StageContext
  ): void {
    if (
      !(
        context.network &&
        context.network.vpcName &&
        context.network.publicSubnets &&
        context.network.publicSubnets.length > 0 &&
        context.network.targetGroupArn &&
        context.network.securityGroup &&
        ((context.network.securityGroup.id === "" &&
          context.network.securityGroup.name) ||
          context.network.securityGroup.id)
      )
    ) {
      throw new Error(
        `error: invalid network context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }
    if (
      !(
        context.container &&
        context.container.hostPort > 0 &&
        context.container.containerPort > 0 &&
        context.container.memoryLimitMiB > 0 &&
        context.container.cpu > 0 &&
        context.container.desiredCount > 0 &&
        context.container.imageName
      )
    ) {
      throw new Error(
        `error: invalid container context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }
  }

  private getNetworkResources(
    env: string,
    network: networkConfig
  ): {
    vpc: ec2.IVpc;
    securityGroup: ec2.ISecurityGroup;
  } {
    // retrieve vpc by name
    const vpc: ec2.IVpc = ec2.Vpc.fromLookup(this, "VPC", {
      vpcId: network.vpcName,
    });

    const securityGroup: ec2.ISecurityGroup = this.getOrCreateSecurityGroup(
      env,
      vpc,
      network
    );

    return { vpc, securityGroup };
  }

  private getOrCreateSecurityGroup(
    env: string,
    vpc: ec2.IVpc,
    network: networkConfig
  ): ec2.ISecurityGroup {
    if (network.securityGroup.id) {
      return ec2.SecurityGroup.fromSecurityGroupId(
        this,
        "SecurityGroup",
        network.securityGroup.id
      );
    }

    const securityGroup: ec2.SecurityGroup = new ec2.SecurityGroup(
      this,
      "SecurityGroup",
      {
        vpc: vpc,
        securityGroupName: network.securityGroup.name,
      }
    );

    securityGroup.addIngressRule(ec2.Peer.ipv4("0.0.0.0/0"), ec2.Port.tcp(80));
    securityGroup.addIngressRule(
      ec2.Peer.ipv4("0.0.0.0/0"),
      ec2.Port.tcp(8080)
    );

    return securityGroup;
  }

  private createECS(
    tag: string,
    container: containerConfig,
    vpc: ec2.IVpc,
    securityGroup: ec2.ISecurityGroup
  ): ecs.Ec2Service {
    const cluster: ecs.Cluster = new ecs.Cluster(this, "Cluster", {
      clusterName: `${this.stackName}`,
      vpc,
    });

    cluster.addCapacity("DfaultAutoScalingGroup", {
      instanceType: ec2.InstanceType.of(
        ec2.InstanceClass.T3,
        ec2.InstanceSize.MICRO
      ),
      machineImage: ecs.EcsOptimizedImage.amazonLinux2(),
    });

    const taskDefinition: ecs.Ec2TaskDefinition = new ecs.Ec2TaskDefinition(
      this,
      `${this.stackName}`,
      {
        networkMode: ecs.NetworkMode.BRIDGE,
      }
    );

    const image: ecs.ContainerImage = ecs.ContainerImage.fromEcrRepository(
      ecr.Repository.fromRepositoryName(
        this,
        "Repository",
        container.imageName
      ),
      tag
    );

    taskDefinition
      .addContainer(container.imageName, {
        image,
        memoryLimitMiB: container.memoryLimitMiB,
        cpu: container.cpu,
        environment: {
          APP_ENV: "Demo",
          TZ: "Asia/Tokyo",
        },
        logging: new ecs.AwsLogDriver({
          streamPrefix: `${this.stackName}`,
        }),
      })
      .addPortMappings({
        hostPort: container.hostPort,
        containerPort: container.containerPort,
      });

    // const securityGroups: ec2.ISecurityGroup[] = [];
    // securityGroups.push(securityGroup);
    const ec2Service: ecs.Ec2Service = new ecs.Ec2Service(this, "EC2Service", {
      serviceName: `${this.stackName}`,
      cluster,
      taskDefinition,
      desiredCount: container.desiredCount,
      assignPublicIp: true,
    });

    return ec2Service;
  }

  private createSecurityGroup(
    network: networkConfig,
    vpc: ec2.IVpc
  ): ec2.SecurityGroup {
    const securityGroup: ec2.SecurityGroup = new ec2.SecurityGroup(
      this,
      network.securityGroup.id,
      {
        allowAllOutbound: true,
        securityGroupName: network.securityGroup.name,
        vpc,
      }
    );
    securityGroup.addIngressRule(ec2.Peer.ipv4("0.0.0.0/0"), ec2.Port.tcp(80));
    securityGroup.addIngressRule(ec2.Peer.ipv4("0.0.0.0/0"), ec2.Port.tcp(443));
    return securityGroup;
  }

  private createAlb(
    alb: albConfig,
    securityGroup: ec2.ISecurityGroup,
    vpc: ec2.IVpc
  ): elbv2.ApplicationLoadBalancer {
    return new elbv2.ApplicationLoadBalancer(this, alb.id, {
      internetFacing: true,
      loadBalancerName: alb.name,
      securityGroup: securityGroup,
      vpc,
      vpcSubnets: { subnets: vpc.publicSubnets },
    });
  }

  private addListener(
    alb: elbv2.ApplicationLoadBalancer,
    listener: listenerConfig,
    targetGroup: targetGroupConfig,
    container: containerConfig,
    ec2Service: ecs.Ec2Service
  ): void {
    const httpListener: elbv2.ApplicationListener = alb.addListener(
      listener.http.id,
      {
        port: listener.http.port,
        open: true,
      }
    );

    httpListener.addTargets("ECS", {
      port: targetGroup.port,
      targets: [
        ec2Service.loadBalancerTarget({
          containerName: container.imageName,
          containerPort: container.containerPort,
        }),
      ],
      healthCheck: {
        interval: cdk.Duration.seconds(60),
        path: "/",
        timeout: cdk.Duration.seconds(5),
      },
    });

    const httpsListener: elbv2.ApplicationListener = alb.addListener(
      listener.https.id,
      {
        port: listener.https.port,
        open: true,
        certificates: [
          {
            certificateArn: listener.https.certificateArn,
          },
        ],
      }
    );

    httpsListener.addTargets("ECS", {
      port: targetGroup.port,
      targets: [
        ec2Service.loadBalancerTarget({
          containerName: container.imageName,
          containerPort: container.containerPort,
        }),
      ],
      healthCheck: {
        interval: cdk.Duration.seconds(60),
        path: "/",
        timeout: cdk.Duration.seconds(5),
      },
    });
  }

  private createARecordWithAlb(
    route53Config: route53Config,
    alb: elbv2.ApplicationLoadBalancer
  ): void {
    new route53.ARecord(this, route53Config.id, {
      recordName: route53Config.fqdn,
      target: route53.RecordTarget.fromAlias(
        new route53Targets.LoadBalancerTarget(alb)
      ),
      zone: route53.HostedZone.fromLookup(this, route53Config.hostedZone.id, {
        domainName: route53Config.hostedZone.domainName,
      }),
    });
  }
}
