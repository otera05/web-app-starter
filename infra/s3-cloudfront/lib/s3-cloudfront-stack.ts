import * as cdk from "@aws-cdk/core";
import * as cloudFront from "@aws-cdk/aws-cloudfront";
import * as s3 from "@aws-cdk/aws-s3";
import * as route53 from "@aws-cdk/aws-route53";
import * as route53Targets from "@aws-cdk/aws-route53-targets";

interface s3Config {
  bucketName: string;
  name: string;
}

interface cloudFrontConfig {
  distributionId: string;
  oaiId: string;
  acmArn: string;
  domainName: string;
}

interface hostedZoneConfig {
  id: string;
  domainName: string;
}

interface route53Config {
  id: string;
  hostedZone: hostedZoneConfig;
}

interface StageContext {
  description: string;
  domainName: string;
  s3: s3Config;
  cloudFront: cloudFrontConfig;
  route53: route53Config;
}

export class S3CloudfrontStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    const env: string = this.node.tryGetContext("env") || "Demo";
    const revision: string = this.node.tryGetContext("revision") || "";

    const context: StageContext =
      env == "Test"
        ? JSON.parse(this.node.tryGetContext(env))
        : this.node.tryGetContext(env);

    this.validateEnvironment(env, revision, context);

    const oai: cloudFront.OriginAccessIdentity = new cloudFront.OriginAccessIdentity(
      this,
      context.cloudFront.oaiId,
      {
        comment: `TodonPartyBucket-${this.stackName}`,
      }
    );

    const bucket: s3.Bucket = this.createS3Bucket(context);

    const distribution: cloudFront.CloudFrontWebDistribution = this.createCloudFront(
      bucket,
      oai,
      context.cloudFront
    );

    this.createARecordWithCloudFront(context, distribution);
  }

  private validateEnvironment(
    env: string,
    revision: string,
    context: StageContext
  ): void {
    if (!(context.s3 && context.s3.bucketName && context.s3.name)) {
      throw new Error(
        `error: invalid s3 context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }
    if (
      !(
        context.cloudFront &&
        context.cloudFront.distributionId &&
        context.cloudFront.oaiId &&
        context.cloudFront.acmArn &&
        context.cloudFront.domainName
      )
    ) {
      throw new Error(
        `error: invalid cloudFront context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }
    if (
      !(
        context.route53 &&
        context.route53.id &&
        context.route53.hostedZone &&
        context.route53.hostedZone.domainName &&
        context.route53.hostedZone.id
      )
    ) {
      throw new Error(
        `error: invalid route53 context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }
  }

  private createS3Bucket(context: StageContext): s3.Bucket {
    const bucket: s3.Bucket = new s3.Bucket(this, context.s3.name, {
      versioned: false,
      bucketName: context.s3.bucketName,
      publicReadAccess: false,
      blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
      removalPolicy: cdk.RemovalPolicy.DESTROY,
    });
    return bucket;
  }

  private createCloudFront(
    bucket: s3.Bucket,
    oai: cloudFront.OriginAccessIdentity,
    config: cloudFrontConfig
  ): cloudFront.CloudFrontWebDistribution {
    const viewerCertificateConfig: cloudFront.ViewerCertificate = {
      aliases: [config.domainName],
      props: {
        acmCertificateArn: config.acmArn,
        minimumProtocolVersion: cloudFront.SecurityPolicyProtocol.TLS_V1_2_2019,
        sslSupportMethod: cloudFront.SSLMethod.SNI,
      },
    };

    const distProps: cloudFront.CloudFrontWebDistributionProps = {
      viewerCertificate: viewerCertificateConfig,
      originConfigs: [
        {
          s3OriginSource: {
            s3BucketSource: bucket,
            originAccessIdentity: oai,
          },
          behaviors: [
            {
              isDefaultBehavior: true,
              compress: false,
            },
          ],
        },
      ],
      viewerProtocolPolicy: cloudFront.ViewerProtocolPolicy.REDIRECT_TO_HTTPS,

      errorConfigurations: [
        {
          errorCode: 403,
          responseCode: 200,
          responsePagePath: "/index.html",
        },
        {
          errorCode: 404,
          responseCode: 200,
          responsePagePath: "/index.html",
        },
      ],
      priceClass: cloudFront.PriceClass.PRICE_CLASS_ALL,
    };

    const distribution: cloudFront.CloudFrontWebDistribution = new cloudFront.CloudFrontWebDistribution(
      this,
      config.distributionId,
      distProps
    );

    return distribution;
  }

  private createARecordWithCloudFront(
    context: StageContext,
    distribution: cloudFront.CloudFrontWebDistribution
  ) {
    const hostedZone: route53.IHostedZone = route53.HostedZone.fromHostedZoneAttributes(
      this,
      "zone",
      {
        zoneName: context.route53.hostedZone.domainName,
        hostedZoneId: context.route53.hostedZone.id,
      }
    );
    return new route53.ARecord(this, context.route53.id, {
      recordName: context.domainName,
      target: route53.RecordTarget.fromAlias(
        new route53Targets.CloudFrontTarget(distribution)
      ),
      // zone: hostedZone,
      zone: route53.HostedZone.fromLookup(this, context.route53.hostedZone.id, {
        domainName: context.route53.hostedZone.domainName,
      }),
    });
  }
}
