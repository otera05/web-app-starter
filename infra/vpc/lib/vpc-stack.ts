import * as ec2 from '@aws-cdk/aws-ec2';
import { DefaultInstanceTenancy, RouterType } from '@aws-cdk/aws-ec2';
import * as cdk from '@aws-cdk/core';

interface vpcConfig {
  name: string;
  cidr: string;
}

interface igwConfig {
  name: string;
}

interface publicSubnetConfig {
  name: string;
  availabilityZone: string;
  cidrBlock: string;
}

interface privateSubnetConfig {
  name: string;
  availabilityZone: string;
  cidrBlock: string;
}

interface subnetConfig {
  public: publicSubnetConfig[];
  private: privateSubnetConfig[];
}

interface StageContext {
  vpc: vpcConfig;
  igw: igwConfig;
  subnet: subnetConfig;
}

export class VpcStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    const env: string = this.node.tryGetContext('env') || 'Demo';
    const revision: string = this.node.tryGetContext('revision') || '';
    const context: StageContext =
      env == 'Test'
        ? JSON.parse(this.node.tryGetContext(env))
        : this.node.tryGetContext(env);

    this.validateEnvironment(env, revision, context);

    const vpc: ec2.Vpc = this.createVpc(context.vpc);

    const igw: ec2.CfnInternetGateway = this.createInternetGateway(
      vpc,
      context.igw
    );

    this.createPublicSubnet(vpc, igw, context.subnet.public);
    this.createprivateSubnet(vpc, context.subnet.private);
  }

  private validateEnvironment(
    env: string,
    revision: string,
    context: StageContext
  ): void {
    if (!(context.vpc && context.vpc.name && context.vpc.cidr)) {
      throw new Error(
        `error: invalid vpc context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }
    if (!context.igw.name) {
      throw new Error(
        `error: invalid igw context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }

    if (!(context.subnet.public.length > 0)) {
      throw new Error(
        `error: invalid subnet context ${JSON.stringify({
          env: env,
          revision: revision,
          context: context,
        })}`
      );
    }
  }

  private createVpc(vpcConfig: vpcConfig): ec2.Vpc {
    return new ec2.Vpc(this, vpcConfig.name, {
      cidr: vpcConfig.cidr,
      defaultInstanceTenancy: DefaultInstanceTenancy.DEFAULT,
      enableDnsHostnames: true,
      enableDnsSupport: true,
      natGateways: 0,
      subnetConfiguration: [],
    });
  }

  private createInternetGateway(
    vpc: ec2.Vpc,
    igwConfig: igwConfig
  ): ec2.CfnInternetGateway {
    const igw: ec2.CfnInternetGateway = new ec2.CfnInternetGateway(
      this,
      igwConfig.name,
      {}
    );
    new ec2.CfnVPCGatewayAttachment(this, igwConfig.name + '-attachemnt', {
      vpcId: vpc.vpcId,
      internetGatewayId: igw.ref,
    });
    return igw;
  }

  private createPublicSubnet(
    vpc: ec2.Vpc,
    igw: ec2.CfnInternetGateway,
    publicSubnetConfig: publicSubnetConfig[]
  ): void {
    publicSubnetConfig.forEach((pubSubCon: publicSubnetConfig) => {
      const publicSubnet: ec2.Subnet = new ec2.Subnet(this, pubSubCon.name, {
        availabilityZone: pubSubCon.availabilityZone,
        vpcId: vpc.vpcId,
        cidrBlock: pubSubCon.cidrBlock,
        mapPublicIpOnLaunch: true,
      });
      // HACK: 各subnetに対して専用のRouteTableが生成される
      publicSubnet.addRoute('route-demo-mini-app', {
        routerId: igw.ref,
        routerType: RouterType.GATEWAY,
      });
    });
  }

  private createprivateSubnet(
    vpc: ec2.Vpc,
    privateSubnetConfig: privateSubnetConfig[]
  ): void {
    privateSubnetConfig.forEach((prvSubCon: privateSubnetConfig) => {
      new ec2.Subnet(this, prvSubCon.name, {
        availabilityZone: prvSubCon.availabilityZone,
        vpcId: vpc.vpcId,
        cidrBlock: prvSubCon.cidrBlock,
      });
    });
  }
}
