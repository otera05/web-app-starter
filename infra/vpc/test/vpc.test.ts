import * as cdk from '@aws-cdk/core';
import { SynthUtils } from '@aws-cdk/assert';
import { VpcStack } from '../lib/vpc-stack';

test('Snapshot Test', () => {
  const account: string = '000000000000';
  const region: string = 'ap-northeast-1';
  const context: any = {
    description: '',
    vpc: {
      name: 'vpc-demo-mini-app',
      cidr: '10.3.0.0/16',
    },
    igw: {
      name: 'igw-demo-mini-app',
    },
    subnet: {
      public: [
        {
          name: 'public-sub-demo-mini-app-a',
          availabilityZone: 'ap-northeast-1a',
          cidrBlock: '10.3.0.0/24',
        },
      ],
    },
  };
  const app = new cdk.App({
    context: { env: 'Test', Test: JSON.stringify(context) },
  });
  const stack = new VpcStack(app, 'TestStack', {
    env: { account, region },
  });
  const cfn = SynthUtils.toCloudFormation(stack);
  expect(cfn).toMatchSnapshot();
});
