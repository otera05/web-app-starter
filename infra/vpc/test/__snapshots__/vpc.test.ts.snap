// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 1`] = `
Object {
  "Resources": Object {
    "igwdemominiapp": Object {
      "Type": "AWS::EC2::InternetGateway",
    },
    "igwdemominiappattachemnt": Object {
      "Properties": Object {
        "InternetGatewayId": Object {
          "Ref": "igwdemominiapp",
        },
        "VpcId": Object {
          "Ref": "vpcdemominiapp9DD3AA91",
        },
      },
      "Type": "AWS::EC2::VPCGatewayAttachment",
    },
    "publicsubdemominiappaRouteTableAssociation0008BF4C": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "publicsubdemominiappaRouteTableFFBBC5AA",
        },
        "SubnetId": Object {
          "Ref": "publicsubdemominiappaSubnet39559004",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "publicsubdemominiappaRouteTableFFBBC5AA": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/public-sub-demo-mini-app-a",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcdemominiapp9DD3AA91",
        },
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "publicsubdemominiappaSubnet39559004": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "10.3.0.0/24",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/public-sub-demo-mini-app-a",
          },
        ],
        "VpcId": Object {
          "Ref": "vpcdemominiapp9DD3AA91",
        },
      },
      "Type": "AWS::EC2::Subnet",
    },
    "publicsubdemominiapparoutedemominiappF88E4C4F": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": Object {
          "Ref": "igwdemominiapp",
        },
        "RouteTableId": Object {
          "Ref": "publicsubdemominiappaRouteTableFFBBC5AA",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "vpcdemominiapp9DD3AA91": Object {
      "Properties": Object {
        "CidrBlock": "10.3.0.0/16",
        "EnableDnsHostnames": true,
        "EnableDnsSupport": true,
        "InstanceTenancy": "default",
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "TestStack/vpc-demo-mini-app",
          },
        ],
      },
      "Type": "AWS::EC2::VPC",
    },
  },
}
`;
